src/Adapters/Auth/apple.js

```diff
@@ -39,7 +39,10 @@ const verifyIdToken = async ({ token, id }, clientID) => {
       'id token is invalid for this user.'
     );
   }
-  const applePublicKey = await getApplePublicKey();
+
+  const decodedToken = jwt.decode(token, { complete: true });
+  const keyId = decodedToken.header.kid;
+  const applePublicKey = await getApplePublicKey(keyId);
   const jwtClaims = jwt.verify(token, applePublicKey, { algorithms: 'RS256' });
```

Comment:
We should probably parse the `"alg"` key from the JWT dict for the used algorithm and pass it in here, to avoid hardcoding `RS256` in case Apple changes this at some point

Reply:
Nice catch! We will open a new PR with these changes.

Reply:
I'll get the new PR submitted today, sorry I broke the source branch for this one working on a different thing.

Reply:
There is an existing issue https://github.com/parse-community/parse-server/pull/6394. I'll get that in shortly so you won't have merge conflicts.